File: typescript/armstrong-numbers/armstrong-numbers.ts
export default class ArmstrongNumbers {
    constructor() {};
    public static isArmstrongNumber(num: number): boolean {
        let result: number = 0;

        const digitsArray: Array<number> = String(num).split('').map(Number);

        digitsArray.forEach((digit: number) => {
            result += Math.pow(digit, digitsArray.length);
        });
        return result === num;
    }
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

  boolean isArmstrongNumber(int numberToCheck) {
    int sum = 0;
    
    String[] stringToCheck = Integer.toString(numberToCheck).split("");
    for(String digit : stringToCheck) {
      sum += Math.pow(Integer.parseInt(digit), stringToCheck.length);
    }

    return numberToCheck == sum;
  }

}
